src/java/org/apache/cassandra/db/aggregation/GroupMaker.java

```diff
@@ -113,28 +132,94 @@ public PkPrefixGroupMaker(ClusteringComparator comparator, int clusteringPrefixS
         @Override
         public boolean isNewGroup(DecoratedKey partitionKey, Clustering<?> clustering)
         {
-            boolean isNew = false;
+            // We are entering a new group if:
+            // - the partition key is a new one
+            // - the last clustering was not null and does not have the same prefix as the new clustering one
+            boolean isNew = !partitionKey.getKey().equals(lastPartitionKey)
```

Comment:
The method `DecoratedKey#getKey` is called twice, here and a few lines below to set the new value of `lastPartitionKey`. Since the implementation of `getKey` is not necessarily trivial, we might want to use a local variable to avoid calling it twice, wdyt? The same would apply to `SelectorGroupMaker#isNewGroup`.

Reply:
Saw this is changed in `SelectorGroupMaker#isNewGroup`, we're still missing it here, in `PkPrefixGroupMaker#isNewGroup`.